### PR TITLE
Dataloader

### PR DESCRIPTION
**Pull request by sbawri**
Added Dataloader and homography changes

### CODE REVIEW

File: dataloaders/datasets/imagry.py [hunk 1]
```diff
@@ -0,0 +1,109 @@
...
+
+        paths = self.lines[index].split(",")
+
+        transform = transforms.Compose([transforms.PILToTensor()])
+        inputs = []
+        for i in range(len(paths)-1):
+            inputs.append(transform(Image.open(paths[i])))  
+        gt = get_gt(paths[-1].strip())
+
+        return torch.stack(inputs), gt
```
Comment (adhagat-imagry) [inline dataloaders/datasets/imagry.py:39]: The `inputs` are a tensor while `gt` is np-array (as is being returned from `get_gt()`). Perhaps make both into tensors?

File: homography/images/1719447568.245852.jpeg
Comment (adhagat-imagry) [inline homography/images/1719447568.245852.jpeg:1]: Could these images be added to an s3 bucket instead?

File: scripts/missing_files.py [hunk 2]
```diff
@@ -0,0 +1,32 @@
...
+    for line in lines:
+        line_split = line.split(",")
+        flag = False
+        for i in range(len(line_split)):
+            if not os.path.isfile(line_split[i].strip()):
+                missing_count[i].append(line_split[i].strip())
+                # print("Missing ", line_split[i].strip())
+                if not flag:
+                    flag = True
+                    count += 1
```
Comment (adhagat-imagry) [inline scripts/missing_files.py:20]: Minor comment - this looks similar to the logic in the dataset file to find / ignore files with missing image(s). Could a create a helper / utility function and reuse. Not urgent and can be done during a refactor too.

**Review by adhagat-imagry — CHANGES_REQUESTED** [review]
Any reason we want images / masks to be on git? Could we move them to s3 instead?

File: homography/images/1719447568.245852.jpeg
Comment (bodsul2024) [reply homography/images/1719447568.245852.jpeg:1]: +1 @sbawri please remove images from this commit and implement functionality to download from say s3.

File: mypath.py [hunk 3]
```diff
@@ -9,6 +9,8 @@ def db_root_dir(dataset):
             return '/path/to/datasets/cityscapes/'     # foler that contains leftImg8bit/
         elif dataset == 'coco':
             return '/path/to/datasets/coco/'
+        elif dataset == 'imagry':
+            return '/home/sumitbawri/data_train.txt' # Path to the text file containing data
```
Comment (bodsul2024) [inline mypath.py:13]: obviously its bad practice to have your name showing up in source code like this

File: scripts/create_dataset.py [hunk 4]
```diff
@@ -0,0 +1,81 @@
+from collections import defaultdict
+
+import argparse
+# import blosc
```
Comment (bodsul2024) [inline scripts/create_dataset.py:4]: rm if not needed

**Review by bodsul2024 — CHANGES_REQUESTED** [review]
Can we not commit images? Makes PR bloated.

Looks good modulo comments. I don't see the training code. I can review again after that is added to the commit.
